agent/test-agent-cli/src/main.rs

```diff
@@ -0,0 +1,88 @@
+mod error;
+mod get_secret;
+mod init;
+mod retry_count;
+mod send_results;
+mod terminate;
+mod test_error;
+
+use argh::FromArgs;
+use env_logger::Builder;
+use error::Result;
+use log::LevelFilter;
+use snafu::ResultExt;
+use test_agent::BootstrapData;
+use test_agent::{Client, DefaultClient};
+
+#[derive(FromArgs)]
+/// This command line interface is to send/receive information from K8s cluster when creating testcase using Bash script
+struct Args {
+    /// set logging verbosity [trace|debug|info|warn|error]. If the environment variable `RUST_LOG`
+    /// is present, it overrides the default logging behavior. See https://docs.rs/env_logger/latest
+    #[argh(option, default = "LevelFilter::Info")]
+    log_level: LevelFilter,
+
+    #[argh(subcommand)]
+    command: Command,
+}
+
+#[derive(FromArgs, PartialEq, Debug)]
+#[argh(subcommand)]
+enum Command {
+    /// Send any error encountered
+    TestError(test_error::TestError),
+    /// Get secret  value for a key
+    GetSecret(get_secret::GetSecret),
+    /// Set the Task state running, return config details
+    Init(init::Init),
+    /// Get number of retries allowed
+    RetryCount(retry_count::RetryCount),
+    /// Send test results
+    SendResults(send_results::SendResults),
+    /// Mark Task state complete, handle keep running, save all results as tar
+    Terminate(terminate::Terminate),
+}
+
+#[tokio::main]
+async fn main() {
+    let args: Args = argh::from_env();
+    init_logger(args.log_level);
+    if let Err(e) = run(args).await {
+        eprintln!("{:?}", e);
+        std::process::exit(1);
+    }
+}
+
+async fn run(args: Args) -> Result<()> {
+    let bootstrap_data = BootstrapData::from_env().unwrap();
+    let client = DefaultClient::new(bootstrap_data)
```

Comment:
This will not work as expected. `DefaultClient::new` creates a new temp_dir in the `new` function so each command will be using a different `results_dir`.